sdk/examples/how_tos/native_tokens/destroy_foundry.rs

```diff
@@ -34,12 +38,58 @@ async fn main() -> Result<()> {
         let foundry_count = balance.foundries().len();
         println!("Foundries before destroying: {foundry_count}");
 
+        let token_id = TokenId::from(*foundry_id);
+
         // Set the stronghold password
         wallet
             .set_stronghold_password(std::env::var("STRONGHOLD_PASSWORD").unwrap())
             .await?;
 
-        let transaction = account.burn(*foundry_id, None).await?;
+        // Check if all tokens are melted.
+        let native_tokens: Vec<&NativeTokensBalance> = balance
+            .native_tokens()
+            .iter()
+            .filter(|native_token| *native_token.token_id() == token_id)
+            .collect();
```

Comment:
I think this should just use `.find()`, so `native_tokens` is `Option<NativeTokensBalance>`, because there can only be one `NativeTokensBalance` for one native token

Reply:
Or actually `if let Some(native_token_balance) = ` is probably even better, so you don't need this `if native_tokens.len() > 0 {` or `-is_some()` after changing

Reply:
yea i thought it could have duplicate token ids, fixed